src/cmt_decode_prometheus.c

```diff
@@ -821,24 +821,26 @@ static int add_metric_summary(struct cmt_decode_prometheus_context *context)
 
 static int finish_metric(struct cmt_decode_prometheus_context *context)
 {
-    int rv;
+    int rv = 0;
 
-    switch (context->metric.type) {
-        case COUNTER:
-            rv = add_metric_counter(context);
-            break;
-        case GAUGE:
-            rv = add_metric_gauge(context);
-            break;
-        case HISTOGRAM:
-            rv = add_metric_histogram(context);
-            break;
-        case SUMMARY:
-            rv = add_metric_summary(context);
-            break;
-        default:
-            rv = add_metric_untyped(context);
-            break;
+    if (mk_list_is_empty(&context->metric.samples) != 0) {
```

Comment:
not an issue but as a good practice to avoid adding a new nested level you can return as soon you check the condition is empty, e.g:

```c
if (mk_list_is_empty(&context->metric.samples) == 0) {
    return 0;
}
```

btw, I think there is no return value when it finish

Reply:
Instead of returning early, I've used a goto to also run the cleanup function